translator/translate/otel/processor/resourcedetection/translator.go

```diff
@@ -55,7 +63,38 @@ func (t *translator) ID() component.ID {
 	return component.NewIDWithName(t.factory.Type(), t.name)
 }
 
-func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+func (t *translator) Translate(*confmap.Conf) (component.Config, error) {
 	cfg := t.factory.CreateDefaultConfig().(*resourcedetectionprocessor.Config)
-	return common.GetYamlFileToYamlConfig(cfg, appSignalsAwsResourceDetectionConfig)
+	cfg.Override = true
+	cfg.Timeout = time.Second * 2
+	c := confmap.NewFromStringMap(map[string]any{
+		"ec2": map[string]any{
+			"tags": []string{
+				"^kubernetes.io/cluster/.*$",
+				"^aws:autoscaling:groupName",
+			},
+		},
+	})
+	if err := c.Unmarshal(&cfg); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal into resource detection processor: %w", err)
+	}
+
+	mode := context.CurrentContext().KubernetesMode()
+	if mode == "" {
+		mode = context.CurrentContext().Mode()
+	}
+	if mode == config.ModeEC2 {
+		if ecsutil.GetECSUtilSingleton().IsECS() {
+			mode = config.ModeECS
+		}
+	}
+
+	switch mode {
+	case config.ModeEKS:
+		cfg.Detectors = []string{detectorTypeEKS, detectorTypeEnv, detectorTypeEC2}
+	default:
+		cfg.Detectors = []string{detectorTypeEnv, detectorTypeEC2}
```

Comment:
Just curious but what happens if ECS type is fargate? Will EC2 detector fail silently and proceed or will it cause a failure due to IMDS being unavailable.